File: weekly_test7/distinct_lines.pl
#!/usr/bin/perl -w

use strict;
my $num = $ARGV[0];
my @input;
my %count;
my $count_num;
my $line_num;
while (my $line = <STDIN>){
        chomp $line;
        $line = lc $line;
        $line =~ s/\s+/ /g;
        $line =~ s/^\s+//g;
        if (!exists $count{$line}){
                $count{$line} = 1;
                $count_num++;
                if ($num == $count_num){
                        $line_num = $.;
                }
        }

}
#print "$count_num\n";
if ($num == $count_num){
        print "$num distinct lines seen after $line_num lines read.\n";
        }
elsif ($num < $count_num){
        print "$num distinct lines seen after $line_num lines read.\n";
}
else{
        print "End of input reached after $. lines read - $num different lines not seen.\n";
}

File: weekly_test9/distinct_lines.py
#!/usr/bin/python3

import sys
import re

lines = []
s = []
num = int(sys.argv[1])
for line in sys.stdin:
        line = line.lower()
        line = re.sub(r"\s+", " ", line)
        line = re.sub(r"^\s+", "", line)
        lines.append(line)

line_num = 0
count = 0
for i in lines:
        line_num = line_num + 1
        if i not in s:
                s.append(i)
                count = count + 1
        if count == num:
                print(str(count) + " distinct lines seen after " + str(line_num) + " lines read.")
                break

if count < num:
        print("End of input reached after "+ str(line_num) + " lines read -  " + str(num) + " different lines not seen.")
